src/rule.ts

```diff
@@ -0,0 +1,24 @@
+export type AttributeValueType = string | number;
+
+export enum OperatorType {
+  MATCHES = 'MATCHES',
+  GTE = 'GTE',
+  GT = 'GT',
+  LTE = 'LTE',
+  LT = 'LT',
+}
+
+export interface Condition {
+  operator: OperatorType;
+  attribute: string;
+  value: AttributeValueType;
+}
+
+export enum RuleType {
+  AND = 'AND', // all conditions evaluate to true
```

Comment:
To me it doesn't look like it would be easy to add a new rule type here, for example `OR`, since the `RuleType` is on the same level as `Condition[]`, which means the type applies to all conditions rather than a pair of conditions. 

If we only support one rule type, what do you think about removing this and assuming that all `condition` evaluations are `AND`ed together?